packages/wrangler/src/__tests__/kv.test.ts

```diff
@@ -1132,6 +1131,151 @@ describe("wrangler", () => {
+          "PUT",
+          ([_url, accountId, namespaceId], { body }) => {
+            requests.count++;
+            expect(accountId).toEqual("some-account-id");
+            expect(namespaceId).toEqual(expectedNamespaceId);
+            expect(JSON.parse(body as string)).toEqual(expectedKeyValues);
+            return null;
+          }
+        );
+        return requests;
+      }
+
+      it("should put the key-values parsed from a file", async () => {
+        const keyValues: KeyValue[] = [
+          { key: "someKey1", value: "someValue1" },
+          { key: "ns:someKey2", value: "123", base64: true },
+          { key: "someKey3", value: "someValue3", expiration: 100 },
+          { key: "someKey4", value: "someValue4", expiration_ttl: 500 },
+        ];
+        writeFileSync("./keys.json", JSON.stringify(keyValues));
+        const requests = mockPutRequest("some-namespace-id", keyValues);
+        await runWrangler(
+          `kv:bulk put --namespace-id some-namespace-id keys.json`
+        );
+        expect(requests.count).toEqual(1);
+      });
+
+      it("should error if the file is not a JSON array", async () => {
+        const keyValues = { key: "someKey1", value: "someValue1" };
+        writeFileSync("./keys.json", JSON.stringify(keyValues));
+        await expect(
+          runWrangler(`kv:bulk put --namespace-id some-namespace-id keys.json`)
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`
+                "Unexpected JSON input from \\"keys.json\\".
+                Expected an array of key-value objects but got object"
+              `);
+      });
+
+      it("should error if the array contains items that are not key-value objects", async () => {
+        const keyValues = [
+          123,
+          "a string",
+          { key: "someKey" },
+          { value: "someValue" },
+          { key: "someKey1", value: "someValue1", invalid: true },
+        ];
+        writeFileSync("./keys.json", JSON.stringify(keyValues));
+        await expect(
+          runWrangler(`kv:bulk put --namespace-id some-namespace-id keys.json`)
+        ).rejects.toThrowErrorMatchingInlineSnapshot(`
+                "Unexpected JSON input from \\"keys.json\\".
+                Each item in the array should be an object that matches:
+
+                interface KeyValue {
+                  key: string;
+                  value: string;
+                  expiration?: number;
+                  expiration_ttl?: number;
+                  metadata?: object;
+                  base64?: boolean;
+                }
+
+                The item at index 0 is a number: 123
+                The item at index 1 is a string: \\"a string\\"
+                The item at index 2 is {\\"key\\":\\"someKey\\"}
+                The item at index 3 is {\\"value\\":\\"someValue\\"}
+                The item at index 4 is {\\"key\\":\\"someKey1\\",\\"value\\":\\"someValue1\\",\\"invalid\\":true}"
```

Comment:
Should we be more lenient and just extract the props that we care about, rather then erroring?

Reply:
So as to not break existing projects, imo it depends on what wrangler1 is doing, which is I assume is being lenient?

Reply:
I'll change this to a warning...